### PR TITLE
Materialize the class features in `@babel/parser`.

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | Yes
| License                  | MIT

The following parser plugins will be materialized in Babel 7.14
- `classProperties`
- `classPrivateProperties`
- `classPrivateMethods`

### CODE REVIEW

File: packages/babel-parser/test/fixtures/es2022/class-properties/new-target-invalid/output.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "type": "File",
   "start":0,"end":19,"loc":{"start":{"line":1,"column":0},"end":{"line":1,"column":19}},
   "errors": [
-    "SyntaxError: `new.target` can only be used in functions. or class properties (1:8)"
+    "SyntaxError: `new.target` can only be used in functions or class properties. (1:8)"
```
Comment (JLHwung) [inline packages/babel-parser/test/fixtures/es2022/class-properties/new-target-invalid/output.json:5]: Also fixed unexpected `.` in the error messages lol.

File: packages/babel-parser/test/fixtures/es2022/class-private-methods/async-generator/options.json [hunk 2]
```diff
@@ -1,3 +0,0 @@
-{
```
Comment (JLHwung) [inline packages/babel-parser/test/fixtures/es2022/class-private-methods/async-generator/options.json]: This commit is created from https://gist.github.com/JLHwung/970c52d63fd3b5ee14e3ada1cd227d33

**Review by nicolo-ribaudo — APPROVED** [review]
(There is a prettier error)

**Comment by JLHwung** [comment]
@nicolo-ribaudo fixed in https://github.com/babel/babel/pull/13098/commits/00f84dbd832cd499106ee38d31a92a75c4935a8e

File: packages/babel-parser/test/fixtures/es2017/async-functions/no-method-asi/output.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
+{
```
Comment (hzoo) [inline packages/babel-parser/test/fixtures/es2017/async-functions/no-method-asi/output.json:1]: oh I would of expected these tests to be duplicated given they used to throw with the enable parser plugin error (and thus wouldn't be testing the syntax but the plugin), but I guess they are unique actually? Like we would of had a test for asi that didn't throw as well, so you could delete this one.

**Review by hzoo — APPROVED** [review]

**Review by kaicataldo — APPROVED** [review]
:tada:

**Pull request merged**